src/typed-geometry/functions/objects/intersection.hh

```diff
@@ -2269,4 +2327,617 @@ template <class ScalarT>
+{
+    return detail::intersection_segment_object_impl(s, c);
+}
+
+// segment3 - ellipse3
+template <class ScalarT>
+[[nodiscard]] constexpr optional<segment<3, ScalarT>> intersection(segment<3, ScalarT> const& s, ellipse<3, ScalarT> const& e)
+{
+    return detail::intersection_segment_object_impl(s, e);
+}
+
+template <class ScalarT>
+[[nodiscard]] constexpr optional<segment<3, ScalarT>> intersection(ellipse<3, ScalarT> const& e, segment<3, ScalarT> const& s)
+{
+    return detail::intersection_segment_object_impl(s, e);
+}
+
+// segment3 - sphere3
+template <class ScalarT>
+[[nodiscard]] constexpr optional<segment<3, ScalarT>> intersection(segment<3, ScalarT> const& s, sphere<3, ScalarT> const& e)
+{
+    return detail::intersection_segment_object_impl(s, e);
+}
+
+template <class ScalarT>
+[[nodiscard]] constexpr optional<segment<3, ScalarT>> intersection(sphere<3, ScalarT> const& e, segment<3, ScalarT> const& s)
+{
+    return detail::intersection_segment_object_impl(s, e);
+}
+
+// segment3 - tube3
+template <class ScalarT>
+[[nodiscard]] constexpr optional<segment<3, ScalarT>> intersection(segment<3, ScalarT> const& s, tube<3, ScalarT> const& t)
+{
+    return detail::intersection_segment_object_impl(s, t);
+}
+
+template <class ScalarT>
+[[nodiscard]] constexpr optional<segment<3, ScalarT>> intersection(tube<3, ScalarT> const& t, segment<3, ScalarT> const& s)
+{
+    return detail::intersection_segment_object_impl(s, t);
+}
+
+// segment3 - cylinder_boundary
+template <class ScalarT>
+[[nodiscard]] constexpr hits<2, tg::pos<3, ScalarT>> intersection(segment<3, ScalarT> const& s, cylinder_boundary<3, ScalarT> const& c)
+{
+    // TODO: This is a standard solution that can be applied to any boundary case
+    auto const line = line3::from_points(s.pos0, s.pos1);
+    auto const params = intersection_parameter(line, c);
+
+    if (!params.any())
+        return {};
+
+    auto const dist = distance(s.pos0, s.pos1);
+    auto n_hits = 0;
+    tg::pos<3, ScalarT> ps[2];
+    for (auto i = 0; i < params.size(); ++i)
+    {
+        auto const t = params[i];
+        if (ScalarT(0) <= t && t <= dist)
+        {
+            ps[n_hits++] = line[t];
+        }
+    }
+    return hits<2, tg::pos<3, ScalarT>>(ps, n_hits);
+}
+
+template <class ScalarT>
+[[nodiscard]] constexpr hits<2, tg::pos<3, ScalarT>> intersection(cylinder_boundary<3, ScalarT> const& c, segment<3, ScalarT> const& s)
+{
+    return intersection(s, c);
+}
+
+template <class ScalarT>
+[[nodiscard]] constexpr bool intersects(box<3, ScalarT> const& a, box<3, ScalarT> const& b)
+{
+    if (a.center == b.center)
+        return true;
+
+    // Separating Axes Theorem
+    auto const axis_check = [](tg::dir<3, ScalarT> d, box<3, ScalarT> const& box_a, box<3, ScalarT> const& box_b) -> bool
+    {
+        float minProja = std::numeric_limits<float>::max();
+        float maxProja = std::numeric_limits<float>::min();
```

Comment:
`std::numeric_limits<float>::min()`  is `1.17549e-38`, which is the smalles POSITIVE float, not the actual minimum finite float. Anyways, this function should use ScalarT and `tg::max<ScalarT>()` and `tg::min<ScalarT>()`, which work as expected

Reply:
did not know that.. revised!